lake/data/writer.go

```diff
@@ -63,38 +62,49 @@ func (o *Object) NewWriter(ctx context.Context, engine storage.Engine, path *sto
 	return w, nil
 }
 
-func (w *Writer) Write(rec *zed.Value) error {
-	key := rec.DerefPath(w.poolKey).MissingAsNull()
+func (w *Writer) Write(val *zed.Value) error {
+	w.count++
+	key := val.DerefPath(w.poolKey).MissingAsNull()
 	if w.seekIndex != nil {
-		if err := w.writeIndex(*key); err != nil {
+		if err := w.writeIndex(key); err != nil {
 			return err
 		}
 	}
-	if err := w.writer.Write(rec); err != nil {
+	if err := w.writer.Write(val); err != nil {
 		return err
 	}
-	w.object.Last.CopyFrom(key)
-	w.count++
+	w.object.To.CopyFrom(key)
 	return nil
 }
 
-func (w *Writer) writeIndex(key zed.Value) error {
+func (w *Writer) writeIndex(key *zed.Value) error {
 	w.seekIndexTrigger += len(key.Bytes)
 	if w.first {
 		w.first = false
-		w.object.First.CopyFrom(&key)
-		w.object.Last.CopyFrom(&key)
-		return w.seekIndex.Write(key, 0, 0)
+		w.object.From.CopyFrom(key)
 	}
-	if w.seekIndexTrigger < w.seekIndexStride || bytes.Equal(key.Bytes, w.object.Last.Bytes) {
+	if w.from == nil {
+		w.from = key.Copy()
+	}
+	w.object.To.CopyFrom(key)
+	if w.seekIndexTrigger < w.seekIndexStride {
 		return nil
 	}
 	if err := w.writer.EndStream(); err != nil {
 		return err
 	}
-	w.seekIndexTrigger = 0
-	pos := w.writer.Position()
-	return w.seekIndex.Write(key, w.count, pos)
+	return w.flushSeekIndex()
+}
+
+func (w *Writer) flushSeekIndex() error {
+	if w.from != nil {
+		w.seekIndexTrigger = 0
+		from := w.from
+		to := w.object.To.Copy()
+		w.from = nil
+		return w.seekIndex.Write(from, to, w.count, uint64(w.writer.Position()))
```

Comment:
On line 77 you remove the count increment and yet this is passed to the `seekindex.Writer`. As I said above I think we can remove the counts since we're nixing the indexer.

Reply:
We'll want the counts for selection vectors from the vcache.